phper/src/functions.rs

```diff
@@ -347,35 +562,70 @@ impl Argument {
         let name = ensure_end_with_zero(name);
         Self {
             name,
+            type_hint: None,
             pass_by_ref: true,
             required: false,
+            nullable: false,
+            default_value: None,
         }
     }
+
+    /// Add a type-hint to the argument
+    pub fn with_type_hint(mut self, type_hint: ArgumentTypeHint) -> Self {
+        self.type_hint = Some(type_hint);
+        self
+    }
+
+    /// Allow type-hint to be nullable
+    pub fn allow_null(mut self) -> Self {
+        self.nullable = true;
+        self
+    }
+
+    /// Argument is required (also see by_*<ref|val>)
+    pub fn required(mut self) -> Self {
+        self.required = true;
+        self
+    }
+
+    /// Argument is optional (also see by_<ref|val>_optional)
+    pub fn optional(mut self) -> Self {
+        self.required = false;
+        self
+    }
+
+    /// Argument default value. Example: "'a-string'", "A_CONST", "42",
+    /// "[0=>'zero']"
+    pub fn with_default_value(mut self, default_value: CString) -> Self {
```

Comment:
to fix: instead of accepting a CString here, accept a String or `impl Into<String>` like nearby functions do...